[VR-12836] upload_completed added in artifact entity

anandjakhaniya opened the pull request:
Jira Ticket: https://vertaai.atlassian.net/browse/VR-12836

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/279

anandjakhaniya committed:
upload_completed added in artifact entity

conradoverta reviewed (commented):
Don't we have to set this flag at some point?

conradoverta reviewed (commented):
More specifically, only when committed. The issue is exactly that the artifact might not be complete when we just log it. Example:
1. log
2. try to push to aws and fail

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/279

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/279

anandjakhaniya committed:
changed points are
1. if we support multi-part upload, then mark it as false during log
2. if we do not support multi-part upload, then mark it as true during log

conradoverta reviewed (changes requested):
We don't seem to be setting `upload_completed` during the final commit call, right?

anandjakhaniya committed:
Merge branch 'master' into cm/VR-12591

# Conflicts:
#	backend/src/main/java/ai/verta/modeldb/entities/ArtifactEntity.java
#	backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/ArtifactHandlerBase.java
#	client/verta/verta/_protos/public/registry/RegistryService_pb2.py
#	client/verta/verta/_protos/public/uac/RoleService_pb2.py
#	protos/gen/go/protos/public/registry/RegistryService.pb.go
#	protos/gen/go/protos/public/uac/RoleService.pb.go
#	protos/gen/python/protos/public/registry/RegistryService_pb2.py
#	protos/gen/python/protos/public/uac/RoleService_pb2.py
#	protos/gen/swagger/protos/public/modeldb/HydratedService.swagger.json
#	protos/gen/swagger/protos/public/registry/RegistryService.swagger.json
#	protos/gen/swagger/protos/public/uac/Authorization.swagger.json
#	protos/gen/swagger/protos/public/uac/Collaborator.swagger.json
#	protos/gen/swagger/protos/public/uac/RoleService.swagger.json
#	protos/protos/public/registry/RegistryService.proto
#	protos/protos/public/uac/RoleService.proto

anandjakhaniya commented:
> We don't seem to be setting `upload_completed` during the final commit call, right?

No, We should have to set it. In current implementation, we are setting  `upload_completed` during `commitMultipartArtifact` https://github.com/VertaAI/modeldb/blob/15017c59c907214e128a6d33f79f0285ef1a8320/backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java#L2647

and for new implementation at here https://github.com/VertaAI/modeldb/blob/15017c59c907214e128a6d33f79f0285ef1a8320/backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/ArtifactHandler.java#L297

conradoverta reviewed (changes requested):
Every pre-existing artifact should have the flag set to `true` as it was already uploaded before. Only new artifacts can have `upload_completed=false`.

anandjakhaniya committed:
Marked upload_completed = true to all artifacts

conradoverta commented on backend/src/main/resources/liquibase/db-changelog-2.0.xml:
```diff
@@ -979,4 +979,24 @@
+            <update tableName="artifact">
+                <column name="upload_completed" valueBoolean="false" />
+                <where>field_type="artifacts" AND upload_completed = 1</where>
+            </update>
```
This rollback is not valid. This is an information losing change, so we can't rollback from it. Let's remove to make that clearer.

anandjakhaniya replied:
Ok, got it. removed rollback tag now.

conradoverta reviewed (approved):
One change. LGTM otherwise

anandjakhaniya committed:
Removed rollback as per review comment
